integration_test.go

```diff
@@ -4,320 +4,386 @@ import (
+			It("should create without error", func() {
+				_, err := k8sClientWithDefaults(gvrResourceGroup).Create(context.TODO(), &azureResourecGroupCRDRequest, metav1.CreateOptions{})
+				Expect(err).To(BeNil())
+			}, 30)
+			It("should eventually create the azure resource group resource in azure", func() {
+				Eventually(func() error {
+					_, err := GetAzureResourceGroup(context.TODO(), resourceGroupName)
+					return err
+				}, time.Second*30, time.Second*5).Should(BeNil())
+			}, 30)
+			It("should eventually have a status.id property set", func() {
+				Eventually(func() bool {
+					var err error
+					azureResourceGroupCRDResponse, err = k8sClientWithDefaults(gvrResourceGroup).Get(context.TODO(), resourceGroupName, metav1.GetOptions{})
+					Expect(err).To(BeNil())
+					status, _ := azureResourceGroupCRDResponse.Object["status"].(map[string]interface{})
+					var ok bool
+					resourceGroupID, ok = status["id"].(string)
+					return ok
+				}, time.Second*30, time.Second*5).Should(BeTrue(), "CRD should have status.id property")
+			}, 30)
+			It("should have a status.id property set to a valid azure resource id", func() {
+				Expect(resourceGroupID).To(MatchRegexp("/subscriptions/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/resourceGroups/" + resourceGroupName))
+			}, 30)
+			It("should store terraform state encrypted in status._tfoperator.tfState", func() {
+				if !EncryptionKeyIsSet() {
+					Skip("encryption key is not set, skipping test")
+				}
+				tfStateString, gotTfState, err := unstructured.NestedString(azureResourceGroupCRDResponse.Object, "status", "_tfoperator", "tfState")
 				Expect(err).To(BeNil())
+				Expect(gotTfState).To(BeTrue(), "CRD should have status._tfoperator.tfState property")
 
-				status, ok := obj.Object["status"].(map[string]interface{})
+				var js string
+				err = json.Unmarshal([]byte(tfStateString), &js)
+				Expect(err).ToNot(BeNil()) // Check state is encrypted as it's not valid JSON
+			}, 30)
+		})
+		When("creating an azure storage account CRD", func() {
+			azureStorageAccountCRDRequest := GetAzureStorageAccountCRD(resourceGroupName, storageAccountName, storageAccountLocation)
+			var azureStorageAccountCRDResponse *unstructured.Unstructured
+			var status map[string]interface{}
+			var storageAccountID string
+			var storageAccountResource resources.GenericResource
+			It("should create without error", func() {
+				_, err := k8sClientWithDefaults(gvrStorageAccount).Create(context.TODO(), &azureStorageAccountCRDRequest, metav1.CreateOptions{})
 				Expect(err).To(BeNil())
-				if !ok {
-					return ""
+			}, 30)
+			It("should eventually create the azure storage account resource in azure", func() {
+				Eventually(func() error {
+					var err error
+					storageAccountResource, err = GetAzureResource(context.TODO(),
+						resourceGroupName,
+						"Microsoft.Storage",
+						"storageAccounts",
+						storageAccountName,
+						"2019-06-01",
+					)
+					return err
```

Comment:
This is even nicer, love the generic `GetAzureResource` method instead of having to new up clients for all the different service types.